src/Storages/MergeTree/MergeTreeSequentialSource.cpp

```diff
@@ -41,7 +41,8 @@ MergeTreeSequentialSource::MergeTreeSequentialSource(
     addTotalRowsApprox(data_part->rows_count);
 
     /// Add columns because we don't want to read empty blocks
-    injectRequiredColumns(storage, storage_snapshot, data_part, columns_to_read);
+    injectRequiredColumns(storage, storage_snapshot, data_part, /*with_subcolumns=*/ false, columns_to_read);
```

Comment:
Do I understand correctly that this is the most imprtant line?
Maybe add a comment explaining - why we should not inject subcolumns as required columns here?

Reply:
Yes. I can add a comment. 

Actually the problem is not exactly that we shouldn't require subcolumns, because we will read only columns for requested  names and there are no subcolumns. But subcolumn can be added as the smallest column in case when all requested columns are missing.